File: app/models/membership_payments.py

from app.extensions import db

class MembershipPayments(db.Model):
    __tablename__ = 'membership_payments'

    payment_id = db.Column(db.BigInteger, primary_key=True, autoincrement=True)
    membership_order_id = db.Column(db.String(100), nullable=False)
    membership_pay_amount = db.Column(db.Integer, nullable=False)
    payment_status = db.Column(db.String(20), nullable=False)
    card_brand = db.Column(db.String(20), nullable=False)
    card_last_four = db.Column(db.String(4), nullable=False)
    membership_history_date = db.Column(db.DateTime, nullable=False)

    #외래키
    membership_id = db.Column(db.BigInteger, db.ForeignKey('membership.membership_id'), nullable=False)

File: src/main/java/com/ticketflow/entity/MembershipPayments.java
package com.ticketflow.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "membership_payments")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MembershipPayments {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "payment_id")
    private Long paymentId;

    @Column(name = "membership_order_id", length = 100, nullable = false)
    private String membershipOrderId;

    @Column(name = "membership_pay_amount", nullable = false)
    private Integer membershipPayAmount;

    @Column(name = "payment_status", length = 20, nullable = false)
    private String paymentStatus;

    @Column(name = "card_brand", length = 20, nullable = false)
    private String cardBrand;

    @Column(name = "card_last_four", length = 4, nullable = false)
    private String cardLastFour;

    @Column(name = "membership_history_date", nullable = false)
    private LocalDateTime membershipHistoryDate;

    @Column(name = "webhook_event_id", length = 100, unique = true)
    private String webhookEventId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "membership_id", nullable = false)
    private Membership membership;
}
